### ISSUE TITLE
Evaluation takes too much time. Is it normal?

### ISSUE THREAD

**Issue by yezhejack** [google/seq2seq#206]
The dev data contains 5000 sequences. 
============================================================
INFO:tensorflow:Creating ZeroBridge in mode=eval
INFO:tensorflow:
ZeroBridge: {}

INFO:tensorflow:Starting evaluation at 2017-05-04-07:43:55
I tensorflow/core/common_runtime/gpu/gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 1080,
 pci bus id: 0000:01:00.0)
W tensorflow/core/framework/op_kernel.cc:993] Out of range: FIFOQueue '_30_dev_input_fn/parallel_read_1/common_queue' is closed an
d has insufficient elements (requested 1, current size 0)
         [[Node: dev_input_fn/parallel_read_1/common_queue_Dequeue = QueueDequeueV2[component_types=[DT_STRING, DT_STRING], timeou
t_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/parallel_read_1/common_queue)]]
W tensorflow/core/framework/op_kernel.cc:993] Out of range: FIFOQueue '_30_dev_input_fn/parallel_read_1/common_queue' is closed an
d has insufficient elements (requested 1, current size 0)
         [[Node: dev_input_fn/parallel_read_1/common_queue_Dequeue = QueueDequeueV2[component_types=[DT_STRING, DT_STRING], timeou
t_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/parallel_read_1/common_queue)]]
W tensorflow/core/framework/op_kernel.cc:993] Out of range: PaddingFIFOQueue '_29_dev_input_fn/batch_queue/padding_fifo_queue' is
closed and has insufficient elements (requested 64, current size 0)
         [[Node: dev_input_fn/batch_queue = QueueDequeueUpToV2[component_types=[DT_INT32, DT_STRING, DT_INT32, DT_STRING], timeout
_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/batch_queue/padding_fifo_queue, dev_input_fn/batch_queue/n)]
]
W tensorflow/core/framework/op_kernel.cc:993] Out of range: PaddingFIFOQueue '_29_dev_input_fn/batch_queue/padding_fifo_queue' is
closed and has insufficient elements (requested 64, current size 0)
         [[Node: dev_input_fn/batch_queue = QueueDequeueUpToV2[component_types=[DT_INT32, DT_STRING, DT_INT32, DT_STRING], timeout
_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/batch_queue/padding_fifo_queue, dev_input_fn/batch_queue/n)]
]
W tensorflow/core/framework/op_kernel.cc:993] Out of range: PaddingFIFOQueue '_29_dev_input_fn/batch_queue/padding_fifo_queue' is closed and has insufficient elements (requested 64, current size 0)
         [[Node: dev_input_fn/batch_queue = QueueDequeueUpToV2[component_types=[DT_INT32, DT_STRING, DT_INT32, DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/batch_queue/padding_fifo_queue, dev_input_fn/batch_queue/n)]]
W tensorflow/core/framework/op_kernel.cc:993] Out of range: PaddingFIFOQueue '_29_dev_input_fn/batch_queue/padding_fifo_queue' is closed and has insufficient elements (requested 64, current size 0)
         [[Node: dev_input_fn/batch_queue = QueueDequeueUpToV2[component_types=[DT_INT32, DT_STRING, DT_INT32, DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/batch_queue/padding_fifo_queue, dev_input_fn/batch_queue/n)]]
W tensorflow/core/framework/op_kernel.cc:993] Out of range: PaddingFIFOQueue '_29_dev_input_fn/batch_queue/padding_fifo_queue' is closed and has insufficient elements (requested 64, current size 0)
         [[Node: dev_input_fn/batch_queue = QueueDequeueUpToV2[component_types=[DT_INT32, DT_STRING, DT_INT32, DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/batch_queue/padding_fifo_queue, dev_input_fn/batch_queue/n)]]
W tensorflow/core/framework/op_kernel.cc:993] Out of range: PaddingFIFOQueue '_29_dev_input_fn/batch_queue/padding_fifo_queue' is closed and has insufficient elements (requested 64, current size 0)
         [[Node: dev_input_fn/batch_queue = QueueDequeueUpToV2[component_types=[DT_INT32, DT_STRING, DT_INT32, DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/batch_queue/padding_fifo_queue, dev_input_fn/batch_queue/n)]]
W tensorflow/core/framework/op_kernel.cc:993] Out of range: PaddingFIFOQueue '_29_dev_input_fn/batch_queue/padding_fifo_queue' is closed and has insufficient elements (requested 64, current size 0)
         [[Node: dev_input_fn/batch_queue = QueueDequeueUpToV2[component_types=[DT_INT32, DT_STRING, DT_INT32, DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](dev_input_fn/batch_queue/padding_fifo_queue, dev_input_fn/batch_queue/n)]]
INFO:tensorflow:Finished evaluation at 2017-05-04-08:59:35
INFO:tensorflow:Saving dict for global step 1001: bleu = 16.43, global_step = 1001, log_perplexity = 1.39028, loss = 1.36901, rouge_1/f_score = 0.746607, rouge_1/p_score = 0.945662, rouge_1/r_score = 0.628931, rouge_2/f_score = 0.45424, rouge_2/p_score = 0.582968, rouge_2/r_score = 0.384094, rouge_l/f_score = 0.354937
WARNING:tensorflow:Skipping summary for global_step, must be a float or np.float32.

**Comment by sld** [comment]
I have similar issue:
INFO:tensorflow:Starting evaluation at 2017-05-04-20:25:54
INFO:tensorflow:Finished evaluation at 2017-05-04-20:43:21
Not so fast(

**Comment by ghost** [comment]
I have the same issue; training is relatively fast, evaluation sometimes seems to take forever:

> INFO:tensorflow:Starting evaluation at 2017-05-05-08:24:36
> INFO:tensorflow:Finished evaluation at 2017-05-05-08:36:39

> INFO:tensorflow:Starting evaluation at 2017-05-05-08:42:20
> INFO:tensorflow:Finished evaluation at 2017-05-05-08:54:32

Anyone any idea how to properly deal with this, except reducing evaluation after x amount of steps? 

What about the warnings btw?

> WARNING:tensorflow:Skipping summary for global_step, must be a float or np.float32.

**Comment by sld** [comment]
Turning off ROUGE metrics in text_metrics.yml helps me to significantly improve evaluation time

**Comment by ghost** [comment]
@sld 

What is a clean way to turn off ROUGE ? Just delete the following classes with ROUGE in the .yml file?

> default_params: &default_params
>   - separator: " "
>   - postproc_fn: "seq2seq.data.postproc.strip_bpe"
> metrics:
>   - class: LogPerplexityMetricSpec
>   - class: BleuMetricSpec
>     params:
>       <<: *default_params
>   **- class: RougeMetricSpec
>     params:
>       <<: *default_params
>       rouge_type: rouge_1/f_score
>   - class: RougeMetricSpec
>     params:
>       <<: *default_params
>       rouge_type: rouge_1/r_score
>   - class: RougeMetricSpec
>     params:
>       <<: *default_params
>       rouge_type: rouge_1/p_score
>   - class: RougeMetricSpec
>     params:
>       <<: *default_params
>       rouge_type: rouge_2/f_score
>   - class: RougeMetricSpec
>     params:
>       <<: *default_params
>       rouge_type: rouge_2/r_score
>   - class: RougeMetricSpec
>     params:
>       <<: *default_params
>       rouge_type: rouge_2/p_score
>   - class: RougeMetricSpec
>     params:
>       <<: *default_params
> rouge_type: rouge_l/f_score**

**Comment by kyleyeung** [comment]
@milanv1 Yeah, I deleted all those lines and it worked.

**Comment by yezhejack** [comment]
@milanv1 it worked
